src/lib/command_line_tests/command_line_tests.ml

```diff
@@ -0,0 +1,62 @@
+open Core
+open Async
+
+(**
+ * Test the basic functionality of the coda daemon and client through the CLI
+ *)
+
+let%test_module "Command line tests" =
+  ( module struct
+    (* executable location relative to src/default/lib/command_line_tests
+
+       dune won't allow running it via "dune exec", because it's outside its
+       workspace, so we invoke the executable directly
+
+       the coda.exe executable must have been built before running the test
+       here, else it will fail
+
+     *)
+    let coda_exe = "../../app/cli/src/coda.exe"
+
+    let start_daemon config_dir port =
+      Process.run_exn ~prog:coda_exe
+        ~args:
+          [ "daemon"
+          ; "-background"
+          ; "-client-port"
+          ; sprintf "%d" port
+          ; "-config-directory"
+          ; config_dir ]
+        ()
+
+    let stop_daemon port =
+      Process.run () ~prog:coda_exe
+        ~args:["client"; "stop-daemon"; "-daemon-port"; sprintf "%d" port]
+
+    let start_client port =
+      Process.run ~prog:coda_exe
+        ~args:["client"; "status"; "-daemon-port"; sprintf "%d" port]
+        ()
+
+    let create_config_directory config_dir =
+      (* create empty config dir to avoid any issues with the default config dir *)
+      let%bind _ = Process.run_exn ~prog:"rm" ~args:["-rf"; config_dir] () in
+      let%map _ = Process.run_exn ~prog:"mkdir" ~args:["-p"; config_dir] () in
+      ()
+
+    let test_background_daemon () =
+      let config_dir = "/tmp/coda-spun-test" in
```

Comment:
Can you use `File_system.with_temp_dir` for this?

Reply:
Yes.